nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/GetAzureEventHub.java

```diff
@@ -312,11 +317,13 @@ public void onScheduled(final ProcessContext context) throws ProcessException, U
             receiverFetchTimeout = null;
         }
 
-        final String connectionString = new ConnectionStringBuilder(new URI("amqps://"+namespace+serviceBusEndpoint), eventHubName, policyName, policyKey).toString();
-        setupReceiver(connectionString);
+        final int numThreads = context.getMaxConcurrentTasks();
+        executor = Executors.newScheduledThreadPool(numThreads);
```

Comment:
Consider having a user defined property for how many threads to give the thread pool that event hubs client uses.  Having it be the same as the number of threads NiFi uses means double of everything.  For most cases this would be unnecessary.  Alternatively this could be added later but good documentation should be provided to the user to understand how the resources are being used.  We offer processor annotations for this which cause this to show up to the end user so they can reason over their selections.

Reply:
Great! It's good to let the user specify the thread number. We prefer to add it later in next pr. And for now, instead of using the context.getMaxConcurrentTasks() that doubles the threads, we can use a default thread number of 4, which will be enough for most cases. We will add description to explaination about the thread pool.